code/modules/mob/mob.dm

```diff
@@ -1009,6 +1008,38 @@
 	set hidden = 1
 	set_face_dir(client.client_dir(WEST))
 
+/mob/verb/shiftnorth()
+	set hidden = TRUE
+	if(!canface())
+		return FALSE
+	if(pixel_y <= (default_pixel_y + 16))
+		pixel_y++
+		is_shifted = TRUE
```

Comment:
```suggestion
	if(custom_pixel_y < 16)
		pixel_y++
		custom_pixel_y++
```
And so on and so forth.

Reply:
Also, I'm guessing hotkey macros are why there's 4 separate procs here, and not one that just takes in a DIR?